runtime/near-vm-runner/src/cache.rs

```diff
@@ -1,25 +1,50 @@
 use crate::errors::IntoVMError;
 use crate::prepare;
-use near_primitives::borsh::BorshSerialize;
+use borsh::{BorshDeserialize, BorshSerialize};
 use near_primitives::hash::CryptoHash;
+use near_vm_errors::CacheError::{DeserializationError, SerializationError, WriteError};
 use near_vm_errors::VMError;
 use near_vm_logic::VMConfig;
 use std::fmt;
 use wasmer_runtime::{compiler_for_backend, Backend};
 use wasmer_runtime_core::cache::Artifact;
 use wasmer_runtime_core::load_cache_with;
+use std::convert::TryFrom;
 
 pub(crate) fn compile_module(
-    _code_and_config_hash: Vec<u8>,
     code: &[u8],
     config: &VMConfig,
 ) -> Result<wasmer_runtime::Module, VMError> {
     let prepared_code = prepare::prepare_contract(code, config)?;
     wasmer_runtime::compile(&prepared_code).map_err(|err| err.into_vm_error())
 }
 
-pub fn get_hash(code: &[u8], config: &VMConfig) -> CryptoHash {
-    near_primitives::hash::hash(&[code, &config.non_crypto_hash().to_le_bytes()].concat())
+#[derive(Debug, Clone, BorshDeserialize, BorshSerialize)]
+enum ContractCacheKey {
+    Version1 {
```

Comment:
Add a comment to explain which wasmer version we use

Reply:
Added VM kind to key explicitly.